BranchMetrics/ios-branch-deep-linking-attribution
# [minor][SDK-1785] Add automatic event logging for IAP

## Reference
SDK-1785 -- [iOS] App Store IAP Event Tracking

## Summary
Add a new public function,`logEventWithTransaction()`, to log in-app purchases and subscriptions as Branch Events. Automatic logging can also be enabled with ` Branch.setLogInAppPurchasesAsEventsEnabled(true)`

## Motivation
Make it easier for customers to log their IAP as Branch Events with much less code.

## Type Of Change


## Testing Instructions
Either add an IAP to a test app, or use the Timber test bed which already has test in-app purchases available. Then try passing a transaction into the new function as well as enabling automatic logging. You should see a v2/event fired after a succesfull purchase.

#### Logging Event with SKPaymentTransaction
```
let event = BranchEvent(name: "")
event.logEvent(with: transaction as! SKPaymentTransaction)
```
#### Enabling Automatic Event Logging
```
Branch.setLogInAppPurchasesAsEventsEnabled(true)
```









cc @BranchMetrics/saas-sdk-devs for visibility.
